### PR TITLE
fix(TCOMP-586): Ensure that ClassLoaders can be garbage collected.

### PR DESCRIPTION
**Pull request by RyanSkraba**
**What is the current behavior?** (You can also link to an open issue here)

We discovered three sources of memory leaks for loaded classes:

1) The AvroRegistry can retain classes as part of shared IndexedRecord converters.  This class has been deprecated in any case and we should not be sharing these converters.

2) The Hadoop FileSystem will automatically create a shutdown hook on a filesystem if it is put into a reusable cache.  Caches can be deactivated with a Hadoop configuration value.

3) The MaxMetaspaceSize JVM option must be set to indicate that Class and ClassLoader objects can be recovered.

**What is the new behavior?**

The three fixes have been implemented.   Tests have been updated for the new code changes, but verifying the behaviour of a long-lived process is not practical in a unit test!

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) ?
- [ ] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/652/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/652/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/654/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/654/)

File: services/components-api-service-rest/scripts/start.sh [hunk 1]
```diff
@@ -41,7 +41,7 @@ fi
 
 APP_CLASS="org.talend.components.service.rest.Application"
 
-JAVA_OPTS="-Xmx2048m -Dfile.encoding=UTF-8 -Dorg.ops4j.pax.url.mvn.localRepository=\"$PWD/.m2\" -Dorg.ops4j.pax.url.mvn.settings=\"$PWD/config/settings.xml\" -Dcomponent.default.config.folder=\"$PWD/config/default\""
+JAVA_OPTS="-XX:MaxMetaspaceSize=250m -Xmx2048m -Dfile.encoding=UTF-8 -Dorg.ops4j.pax.url.mvn.localRepository=\"$PWD/.m2\" -Dorg.ops4j.pax.url.mvn.settings=\"$PWD/config/settings.xml\" -Dcomponent.default.config.folder=\"$PWD/config/default\""
```
Comment (sgandon) [inline services/components-api-service-rest/scripts/start.sh]: Could you also change it on the .bat file please.

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/655/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/655/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/656/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/656/)

**Comment by sgandon** [comment]
@RyanSkraba, once you reviewed it and eventually approuved it, I'll take care of the merge and cherry-pick on the maintenance branch.

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/657/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/657/)

**Pull request merged**